helm-chart/mezod/templates/statefulset.yaml

```diff
@@ -0,0 +1,149 @@
+      securityContext:
+        runAsUser: {{ .Values.securityContext.userId }}
+        runAsGroup: {{ .Values.securityContext.groupId }}
+        fsGroup: {{ .Values.securityContext.groupId }}
+        fsGroupChangePolicy: "OnRootMismatch"
+      {{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName | quote }}
+      {{- end }}
+      volumes:
+        - name: dummy-data
+          emptyDir: {}
+        {{- if .Values.customConfigs.enabled }}
+        - name: custom-configs
+          configMap:
+            name: "custom-configs-{{ include "this.name" . }}"
+        {{- end }}
+      containers:
+        - name: mezod
+          image: "{{ .Values.image }}:{{ .Values.tag }}"
+          imagePullPolicy: IfNotPresent
+          {{- if .Values.maintenanceMode }}
+          args:
+            - /bin/sh
+            - -c
+            - 'trap "exit" TERM; while true; do sleep 1; done'
+          {{- else }}
+          args:
+            - mezod
+            - start
+            - --home={{ .Values.env.MEZOD_HOME }}
+            - --metrics
+            - --log_format={{ .Values.env.MEZOD_LOG_FORMAT }}
+            - --log_level={{ .Values.env.MEZOD_LOG_LEVEL }}
+            - --log_no_color
+          {{- end }}
+          env:
+            {{- range $key, $value := .Values.env }}
+            - name: {{ $key }}
+              value: {{ $value }}
+            {{- end }}
+            - name: KEYRING_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.secrets.keyring | quote }}
+                  key: KEYRING_NAME
+            - name: KEYRING_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.secrets.keyring | quote }}
+                  key: KEYRING_PASSWORD
+            - name: KEYRING_MNEMONIC
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.secrets.keyring | quote }}
+                  key: KEYRING_MNEMONIC
+          ports:
+            {{- range $key, $value := .Values.service.public.ports }}
+            - name: {{ $key }}
+              containerPort: {{ $value }}
+              protocol: TCP
+            {{- end }}
+            {{- range $key, $value := .Values.service.private.ports }}
+            - name: {{ $key }}
+              containerPort: {{ int $value.number }}
+              protocol: TCP
+            {{- end }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          securityContext:
+            capabilities:
+              drop:
+                - ALL
+            allowPrivilegeEscalation: false
+          volumeMounts:
+            - name: data
+              mountPath: {{ .Values.env.MEZOD_HOME | quote }}
+            {{- if .Values.customConfigs.enabled }}
+            - name: custom-configs
+              mountPath: /config/
+            {{- end }}
+        - name: ethereum-sidecar
```

Comment:
Curious why we run the Ethereum sidecar within the same pod. Are there particular benefits over running it as a separate deployment? I know in K8S world the "sidecar" word suggests running it within the same pod but we should rather consider it at a higher abstraction level, something like: "an auxiliary process to mezod client". I think staking providers running multiple validators may benefit if the sidecar is deployed separately as multiple validators could re-use the same instance. Such a setup with sidecar built-in within mezod pod is probably more convoluted.

Reply:
If we want to do that, I'd suggest to do it in a separate PR.

Reply:
I'm fine with the current setup. Just wanted to clarify what was behind the decision. Let's see how it evolves and decide in future.